firetower

```diff
@@ -195,24 +195,29 @@ function get_flag() {
 
 function should_preserve_scrollback() {
   for var in "$@" ; do
-    if [[ $var == --preserve-scrollback ]] ; then
+    if [[ $var == --preserve-scrollback ]] || [[ $var == -p ]] ; then
       return 0
     fi
   done
   return 1
 }
 
 function get_directory() {
-  for var in "$@"
-  do
-    if [[ $var == --directory=* ]] ; then
-      echo "$var" | cut -d '=' -f 2
-      return
-    fi
+  for (( i=1; i<=$#; i++)); do
+    j=$((i+1))
+     if ( [[ ${!i} == -d ]] && [[ ${!j} == * ]] ) ; then
+	echo "${!j}"
+	return
+     elif ( [[ ${!i} == --directory=* ]] ) ; then
+	echo "${!i}" | cut -d '=' -f 2
+	return
+     fi
```

Comment:
Instead of expanding multiple times, would you like to use variables to make it more readable? E.g.
```
local i
local flag
local potential_dir
for (( i=1; i<=$#; i++)); do
  flag="${@:i:1}"
  potential_dir="${@:i+1:1}"
  if ...
```

I know it's simple logic, I'd really like to have unit tests on functions mapping `input -> output` ([here](https://github.com/mweitzel/firetower/blob/master/test)

Also, make sure to keep the indentation consistent, with two additional spaces per indentation. The first line is indented properly but the following are a mismatch of tabs and spaces.